### PR TITLE
Bug Reporting 

### PR DESCRIPTION
**Pull request by Niranjan-B**

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        bugTypeSpinner.setOnItemSelectedListener(this);
+        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        bugTypeSpinner.setAdapter(spinnerAdapter);
+
+        return view;
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        Log.d("bugtype", "" + parent.getItemAtPosition(position).toString());
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: We will be making the API call here?

Comment (Niranjan-B) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Not here, I'll make it on button click. I haven't initialised edit-text nor button.

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Just create the appropriate variables for bug type & message, and assign them the values.
We'll add the API call later.

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Do that also, everything should be ready other than the call part.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        BugReportFragment fragment = new BugReportFragment();
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_bug_report, container, false);
+
+        Spinner bugTypeSpinner = view.findViewById(R.id.spinner_bug_type);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Use butterknife

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java (hunk 1 above)
Comment (Niranjan-B) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Sure, will do it after a couple of hours. It's past midnight here.

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: haha! Sure. no problem 😆

**Comment by Swati4star** [comment]
The API is also ready for this issue. You can incorporate that also.
Check here:
https://github.com/project-travel-mate/server/issues/13#issuecomment-399738611

File: Android/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -163,4 +164,15 @@
     <string name="positive_button">OK</string>
     <string name="negative_button">CANCEL</string>
 
+    <!-- Bug Types -->
+    <array name="bug_types">
+        <item>Crash</item>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: Add `Choose feature type` on top.

**Comment by Niranjan-B** [comment]
@Swati4star , pushed the latest changes.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                displayToast("Failed to send issue...");
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (response.isSuccessful()) {
+                    // will be true only if the status code is in the range of [200..300)
+                    Log.d("RESPONSE : ", "success" + response.toString());
+                    hideProgressDialog();
+                    displayToast("Issue Sent Successfully...");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Add these constant strings to strings.xml file

File: Android/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -164,4 +165,15 @@
     <string name="positive_button">OK</string>
     <string name="negative_button">CANCEL</string>
 
+    <!-- Choose feature type -->
+    <array name="bug_types">
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: Add first string as `Choose issue type`

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+        setupSpinner();
+        mAuthToken = getAuthToken();
+
+        return view;
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        mType = parent.getItemAtPosition(position).toString();
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: show a toast message to `Select an issue` type, if index is 0

**Comment by Swati4star** [comment]
Can you post a screenshot of the screen?

**Comment by Swati4star** [comment]
@Niranjan-B I checked it locally. The screen looks good. 
Just move the string `"Issue Sent Successfully..."` to strings.xml
You can squash the commits into one commit & I'll merge the PR.

**Comment by Niranjan-B** [comment]
@Swati4star , thanks for checking. Will do that.

**Comment by Swati4star** [comment]
Squash that commit into one.
You can rebase your branch with master. You won't have merge commit then.

**Pull request closed without merging**